clang/lib/CIR/FrontendAction/CIRGenAction.cpp

```diff
@@ -203,15 +204,16 @@ class CIRGenConsumer : public clang::ASTConsumer {
 
       // Setup and run CIR pipeline.
       std::string passOptParsingFailure;
-      if (runCIRToCIRPasses(
-              mlirMod, mlirCtx.get(), C, !feOptions.ClangIRDisableCIRVerifier,
-              feOptions.ClangIRLifetimeCheck, lifetimeOpts,
-              feOptions.ClangIRIdiomRecognizer, idiomRecognizerOpts,
-              feOptions.ClangIRLibOpt, libOptOpts, passOptParsingFailure,
-              codeGenOptions.OptimizationLevel > 0,
-              action == CIRGenAction::OutputType::EmitCIRFlat,
-              action == CIRGenAction::OutputType::EmitMLIR, enableCCLowering,
-              feOptions.ClangIREnableMem2Reg)
+      if (runCIRToCIRPasses(mlirMod, mlirCtx.get(), cgm, C,
```

Comment:
I'm still very afraid of all the passes that run before and after lowering prepare and the state of CGM, I don't remember what we ended up discussing, my take is that if we need CGM, we should probably be lowering more during CIRGen and adding more attributes or operations we can do more minimal work later on during lowering prepare. Can you add a quick list of the things you are using CGM for so I can give you extra ideas?

Reply:
Yup, going to play with all the `cgm/cgf` references to see if I can do without it in my next pass. 

FWIW our previous agreement was to do a stripped CGM. e.g. maybe creating a new from the old one without the state and/or a base class that didn't contain the state. I'm not sure we need that after investigating some, but it's still an option.